cobalt/browser/cobalt_content_browser_client.cc

```diff
@@ -27,14 +28,73 @@
 // TODO(b/390021478): Remove this include when CobaltBrowserMainParts stops
 // being a ShellBrowserMainParts.
 #include "content/shell/browser/shell_browser_main_parts.h"
+#include "services/network/public/mojom/network_context.mojom.h"
+#include "starboard/configuration_constants.h"
+#include "starboard/system.h"
 #include "third_party/blink/public/common/web_preferences/web_preferences.h"
+#include "third_party/icu/source/common/unicode/uloc.h"
 
 #if BUILDFLAG(IS_ANDROIDTV)
 #include "cobalt/browser/android/mojo/cobalt_interface_registrar_android.h"
 #endif
 
+#if BUILDFLAG(IS_LINUX)
+#include "components/os_crypt/sync/key_storage_config_linux.h"
+#include "components/os_crypt/sync/os_crypt.h"
+#endif
+
 namespace cobalt {
 
+namespace {
+
+const base::FilePath::CharType kCacheDirname[] = FILE_PATH_LITERAL("Cache");
+const base::FilePath::CharType kCookieFilename[] = FILE_PATH_LITERAL("Cookies");
+const base::FilePath::CharType kNetworkDataDirname[] =
+    FILE_PATH_LITERAL("Network");
+const base::FilePath::CharType kNetworkPersistentStateFilename[] =
+    FILE_PATH_LITERAL("Network Persistent State");
+const base::FilePath::CharType kSCTAuditingPendingReportsFileName[] =
+    FILE_PATH_LITERAL("SCT Auditing Pending Reports");
+const base::FilePath::CharType kTransportSecurityPersisterFilename[] =
+    FILE_PATH_LITERAL("TransportSecurity");
+const base::FilePath::CharType kTrustTokenFilename[] =
+    FILE_PATH_LITERAL("Trust Tokens");
+
+base::FilePath GetCacheDirectory() {
+  std::vector<char> path(kSbFileMaxPath, 0);
+  if (!SbSystemGetPath(kSbSystemPathCacheDirectory, path.data(),
+                       kSbFileMaxPath)) {
+    NOTREACHED() << "Failed to get cache directory.";
+    return base::FilePath();
+  }
+  return base::FilePath(path.data());
+}
+
+std::string GetSystemLanguage() {
+  char buffer[ULOC_LANG_CAPACITY];
+  UErrorCode icu_result = U_ZERO_ERROR;
+
+  // Get the ISO language and country and assemble the system language.
+  uloc_getLanguage(NULL, buffer, std::size(buffer), &icu_result);
+  if (!U_SUCCESS(icu_result)) {
+    DLOG(FATAL) << __FUNCTION__ << ": Unable to get language from ICU for "
+                << "default locale " << uloc_getDefault() << ".";
+    return "en-US";
+  }
+
+  std::string language = buffer;
+  uloc_getCountry(NULL, buffer, std::size(buffer), &icu_result);
+  if (U_SUCCESS(icu_result) && buffer[0]) {
+    language += "-";
+    language += buffer;
+  }
+
+  // We should end up with something like "en" or "en-US".
+  return language;
```

Comment:
It's strange that we need to do all that manipulation ourselves. Can we use some method in e.g. https://source.corp.google.com/h/github/youtube/cobalt/+/main:ui/base/l10n/l10n_util.h to get the locale?

Reply:
Ah I knew it rang a bell: `base::i18n::GetConfiguredLocale()` should do the trick.